src/zgamma_producer.cpp

```diff
@@ -195,7 +221,110 @@ void ZGammaVarProducer::WriteZGammaVars(nano_tree &nano, pico_tree &pico, vector
+            idx_fsr1 = idx_fsr; continue;
+        }
+        if(pico.out_fsrphoton_muonidx()[idx_fsr]==idx_l2){
+          if(idx_fsr2 != -1 && (pico.out_fsrphoton_droveret2()[idx_fsr2] <  pico.out_fsrphoton_droveret2()[idx_fsr])){ continue;}
+            idx_fsr2 = idx_fsr; continue;
+          }
+        }
+
+      //Adds the FSR photons to a map
+      fsrphotons_map.clear();
+      int fsr_cnt = 0;
+      if(idx_fsr1!=-1){
+        fsrphoton1.SetPtEtaPhiM( pico.out_fsrphoton_pt()[idx_fsr1],pico.out_fsrphoton_eta()[idx_fsr1],pico.out_fsrphoton_phi()[idx_fsr1],0 );
+        fsrphotons_map[fsr_cnt] = fsrphoton1;
+        fsr_cnt++;
+      }
+      if(idx_fsr2!=-1){
+        fsrphoton2.SetPtEtaPhiM( pico.out_fsrphoton_pt()[idx_fsr2],pico.out_fsrphoton_eta()[idx_fsr2],pico.out_fsrphoton_phi()[idx_fsr2],0 ); 
+        fsrphotons_map[fsr_cnt] = fsrphoton2;     
+      } 
+    } else {
+      l1.SetPtEtaPhiM(pico.out_el_pt()[idx_l1], pico.out_el_eta()[idx_l1],
+                      pico.out_el_phi()[idx_l1], 0.000511);
+      l2.SetPtEtaPhiM(pico.out_el_pt()[idx_l2], pico.out_el_eta()[idx_l2],
+                      pico.out_el_phi()[idx_l2], 0.000511);
+
+      leptons_map[0] = l1;
+      leptons_map[1] = l2; 
+      leptons_pterr_map[0] = pico.out_el_energyErr()[idx_l1]*l1.Pt()/l1.P();
+      leptons_pterr_map[1] = pico.out_el_energyErr()[idx_l2]*l2.Pt()/l2.P();
+    }
+
+      kinZfitter->Setup(leptons_map, fsrphotons_map, leptons_pterr_map);
+      kinZfitter->KinRefitZ1();
+      refit_leptons = kinZfitter->GetRefitP4s();
+      ll_refit     = refit_leptons[0] + refit_leptons[1];
+      status       = kinZfitter -> GetStatus();
+      covmatstatus = kinZfitter -> GetCovMatStatus();
+      minnll       = kinZfitter -> GetMinNll();
+
+      //debug statement
+      //cnt_refit++; std::cout << cnt_refit << std::endl;
+
+      photon.SetPtEtaPhiM(pico.out_photon_pt()[0], pico.out_photon_eta()[0], pico.out_photon_phi()[0], 0);
+      TLorentzVector llg_refit = ll_refit + photon;
+
+      //Assigns the refit lepton/dilepton quantities
+      pico.out_ll_refit_pt()    = ll_refit.Pt();
+      pico.out_ll_refit_eta()   = ll_refit.Eta();
+      pico.out_ll_refit_phi()   = ll_refit.Phi();
+      pico.out_ll_refit_m()     = ll_refit.M();
+      pico.out_ll_refit_l1_pt() = refit_leptons[0].Pt();
+      pico.out_ll_refit_l2_pt() = refit_leptons[1].Pt();
+
+      pico.out_ll_refit_status()        = status;
+      pico.out_ll_refit_covmat_status() = covmatstatus;
+      pico.out_ll_refit_minnll()        = minnll;
+
+      pico.out_llphoton_refit_pt()   = llg_refit.Pt();
+      pico.out_llphoton_refit_eta()  = llg_refit.Eta();
+      pico.out_llphoton_refit_phi()  = llg_refit.Phi();
+      pico.out_llphoton_refit_m()    = llg_refit.M();
+      pico.out_llphoton_refit_dr()   = ll_refit.DeltaR(photon);
+      pico.out_llphoton_refit_dphi() = ll_refit.DeltaPhi(photon);
+      pico.out_llphoton_refit_deta() = fabs(ll_refit.Eta() - photon.Eta());
```

Comment:
Many event variables are related with the leptons. I wonder if we want to have additional recalculated event variables.
- photon_drmin
- photon_drmax
- llphoton_cosTheta
- llphoton_costheta
- llphoton_psi
- refit_leptons (eta, phi)
- ptt

Also the dijet variables get modified
- llphoton_dijet_dphi
- llphoton_dijet_balance

It would be nice to see how different things become, before doing the production.

Reply:
I will check this in a sample file. I dont imagine they will change a lot, but should I update these variables with the refit value or should there be new variables?

Reply:
Don't know if this is resolved yet, but shouldn't photon_drmin/photon_drmax/lep_eta/lep_phi be unchanged? I was under the impression that the refit only changes lepton pT (and thus things calculated from them including the llγ system angles etc.)

Reply:
That is true. I crossed off them from the list.

Reply:
I think we should have both the refit value and the pre-fit value. The pre-fit value could be useful for synchronization.